src/battle_main.c

```diff
@@ -4418,7 +4419,10 @@ static void HandleTurnActionSelectionState(void)
     {
         RecordedBattle_CheckMovesetChanges(B_RECORD_MODE_RECORDING);
 
-        if (WILD_DOUBLE_BATTLE && gBattleStruct->throwingPokeBall) {
+        if (WILD_DOUBLE_BATTLE
+            && gBattleStruct->throwingPokeBall
+            && gChosenActionByBattler[B_POSITION_PLAYER_RIGHT] != B_ACTION_NOTHING_FAINTED)
+        {
```

Comment:
This change looks good to me, I've confirmed it works locally.

I think for safety we should really use `GetBattlerAtPosition(B_POSITION_PLAYER_RIGHT)` (and `GetBattlerAtPosition(B_POSITION_PLAYER_LEFT)` on the line below). `src/battle_controllers.c` doesn't guarantee that `gBattlerPositions[i] == i` although atm it seems it always is for non-link, non-recorded battles.